🐛  feat/#18 implement a health check endpoint 🐛

DryDish opened the pull request:
hello please tell me what i did wrong, ktnxbey

____
:heavy_check_mark:   Closes:#18

DryDish committed:
test: 💍 adding test file to test commit

DryDish committed:
chore: 🤖 remove previously added test file

DryDish committed:
feat: 🎸 Add '/healthcheck' endpoint and its test

The testing framework works, however the application must be running for
the test to pass

✅ Closes: #18

DryDish commented:
issue: our CI is not testing my test, its only doing the context loads.

also: please turn off download texts from the test printout, that thing is longer than the declaration of independence.

DryDish committed:
fix: 🐛 Moved tests to where they would be run by 'nx test api'

✅ Closes: 18

DryDish commented:
Fixed the test issue, was caused by folder structure. The printout is still useless though.

DryDish commented:
The test now fails because the application is not running, not entirely sure how to fix that since this is a  healthcheck and that means i need to test the live endpoint, mocking it would make 0 sense.
its seen on linenumber ELEVENTHOUSAND!
![image](https://user-images.githubusercontent.com/57399961/111044131-f94bbe80-8446-11eb-9d16-9b325ae1f459.png)

Kwandes commented on apps/api/pom.xml:
```diff
@@ -12,24 +12,58 @@
 
 	<properties>
 		<java.version>11</java.version>
+        <groovy.version>3.0.2</groovy.version> <!-- required for rest-assured or it defaults to an oder one -->
```
Please explain why you need to add groovy to the project

Also, there is a typo
```suggestion
<groovy.version>3.0.2</groovy.version> <!-- required for rest-assured or it defaults to an other one -->
```

Kwandes commented on apps/api/src/test/java/dk/treecreate/api/healthcheckControllerTests/HealthcheckControllerTests.java:
```diff
@@ -0,0 +1,54 @@
+    private static RequestSpecification spec;
+
+
+    // Declared port here, can be changed to be the .properties version eventually
```
If something will end up in the application.properties you should put it in there instead of adding a todo

Kwandes commented on apps/api/src/test/java/dk/treecreate/api/healthcheckControllerTests/HealthcheckControllerTests.java:
```diff
@@ -0,0 +1,54 @@
+package dk.treecreate.api.healthcheckControllerTests;
+
+import dk.treecreate.api.ApiApplication;
+import io.restassured.builder.RequestSpecBuilder;
```
I don't think using Rest Assured library is needed here, you could achieve the same with the built-in MockMvc package

We can discuss this during the MMM

Kwandes commented on .editorconfig:
```diff
@@ -4,7 +4,7 @@ root = true
 charset = utf-8
 indent_style = space
-indent_size = 2
+indent_size = 4
```
Changing this affects the entire project, including the javascript/typescript code
This change, along with adjustment of the formatting, would be its own issue. And we will discuss this during the MMM
```suggestion
indent_size = 2
```

Kwandes commented on apps/api/pom.xml:
```diff
@@ -12,24 +12,58 @@
 			<scope>test</scope>
 		</dependency>
-	</dependencies>
+        <dependency>
```
JUnit is included in the spring-boot-starter-test dependency, no need to include it explicitly

Kwandes commented on apps/api/src/test/java/dk/treecreate/api/healthcheckControllerTests/HealthcheckControllerTests.java:
```diff
@@ -0,0 +1,54 @@
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static io.restassured.RestAssured.*;
+
```
Remove the extra newlines

Kwandes reviewed (dismissed):
Overall, I do not agree with your choice of a new dependency (Rest Assured) instead of using the built-in MockMvc

Details and other problems are posted below

Oh, and make sure to fix the checks so everything passes

Kwandes commented:
Update the pull request title to match our conventions (add an icon etc)
![image](https://user-images.githubusercontent.com/22862227/111028305-11472200-83f6-11eb-8ef7-bcfa1a1a8898.png)

Write in the description what this pull request does, and specify which issue it closes.

DryDish commented:
@Kwandes ignore the re request, not ready yet. :bug:

DryDish committed:
refactor: 💡 Revert two commits to prepare mock tests

Kwandes committed:
Merge branch 'development' into feat/#18-implement-a-health-check-endpoint

DryDish committed:
feat: 🎸 Add healthcheck endpoint and its associated mock test

✅ Closes: #18

DryDish committed:
fix: 🐛 Change status 200 to OK

✅ Closes: #18

DryDish committed:
fix: 🐛 change the enpoint to actually return "OK"

✅ Closes: #18

DryDish committed:
feat: 🎸 Add new healthcheck controller DTO and add examples

A new DTO class had to be created in order to be able to add examples to
the swagger UI documentation. It also includes the model now, so
everyone can see the type.

DryDish committed:
style: 💄 reformat to correct specification of bracket spacing

DryDish committed:
Add a suppress warning for the missing UUID long of custom exception
